src/main/java/hudson/plugins/audit_trail/LogFileDailyRotationAuditLogger.java

```diff
@@ -0,0 +1,172 @@
+                Matcher matcher = Pattern.compile("[0-9]{4}-[0-9]{2}-[0-9]{2}").matcher(lastFile.getName());
+                if (matcher.find()) {
+                    // Initialize initInstant with the date saved on the audit file name
+                    // See example https://stackoverflow.com/questions/14385834/java-regex-group-0
+                    initInstant = Instant.parse(matcher.group(0) + "T00:00:00.00Z").atZone(ZoneId.systemDefault());
+                }
+            }
+        }
+        // Initialize initInstant to the current instant
+        if (initInstant == null) {
+            initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        }
+        configure();
+    }
+
+    String computePattern() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd").withZone(ZoneId.systemDefault());
+        String formattedInstant = formatter.format(initInstant);
+        String computedFileName = String.format("%s-%s", FilenameUtils.getName(basePattern.toString()) , formattedInstant);
+        Path parentFolder = basePattern.getParent();
+        if (parentFolder != null) {
+            return parentFolder.resolve(computedFileName).toString();
+        }
+        return computedFileName;
+    }
+
+    private boolean shouldRotate() {
+        return ZonedDateTime.now().isAfter(initInstant.plus(Duration.ofDays(1)));
+    }
+
+    /**
+     * Rotates the daily rotation logger
+     */
+    private void rotate() {
+        if (getHandler() != null) {
+            getHandler().close();
+        }
+        initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        configure();
+        // After rotating remove old files
+        removeOldFiles();
+    }
+
+    private void removeOldFiles() {
+        Path directoryPath = basePattern.getParent();
+        if (directoryPath != null) {
+            Collection<File> files = FileUtils.listFiles(directoryPath.toFile(), new RegexFileFilter(".*" + FilenameUtils.getName(basePattern.toString()) + DAILY_ROTATING_FILE_REGEX_PATTERN), DirectoryFileFilter.DIRECTORY);
+            if (files.size() > getCount()) {
+                List<File> orderedList = files.stream().sorted(LASTMODIFIED_REVERSE).collect(Collectors.toList());
+                List<File> toDelete = orderedList.subList(getCount(), orderedList.size());
+                for (File file : toDelete) {
+                    if (!file.delete()) {
+                        LOGGER.log(Level.SEVERE, "File {0} could not be removed on rotate overation", file.getName());
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void log(String event) {
+        // to avoid synchronizing the whole method
+        if (shouldRotate()) {
+            synchronized (this) {
+                if (shouldRotate()) rotate();
+            }
+        }
+        super.log(event);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
This is implicit in Java. Perhaps you wanted to add the `initInstant` to the comparison?